catalog/imaging/imaging-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,286 @@
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling bad input during update requests"() {
+        when: "the plugin receives a null input for an update request"
+            nitfPostProcessPlugin.processUpdate(null)
+        then: "the plugin should throw an exception"
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling bad input during delete requests"() {
+        when: "the plugin receives a null input for a delete request"
+            nitfPostProcessPlugin.processDelete(null)
+        then: "the plugin should throw an exception"
+            thrown(IllegalArgumentException)
+    }
+
+    def "handling basic create requests"() {
+        when: "a valid process create request is submitted"
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            2 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value != null})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+            1 * processCreateItem.markMetacardAsModified()
+            createProcessRequest.getProcessItems().size() == 3
+    }
+
+    def "handling processing large files"() {
+        setup:
+            nitfPostProcessPlugin.setMaxNitfSizeMB(4)
+        when: "a request contains a resource that is too large to process"
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then: "nothing should be added to the metacard"
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling requests with overview disabled"() {
+        setup:
+            nitfPostProcessPlugin.setCreateOverview(false)
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("original")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling requests with original disabled"() {
+        setup:
+            nitfPostProcessPlugin.setStoreOriginalImage(false)
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("overview")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling non-nitf resources"() {
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            processResource.getMimeType() >> "text/xml"
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling metacards containing resource uri attribute"() {
+        setup:
+            def mockAttribute = Mock(AttributeImpl)
+        when: "a metacard has an existing resource uri attribute"
+            mockAttribute.name >> "fake"
+            mockAttribute.values >> []
```

Comment:
Should we put a value in here and assert its still there?

Reply:
Done